FileWriteFlags.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace FileWriteFlag
+{
+	// Default values are Do not overwrite files, append to existing files, and write in binary format.
+	enum FileWriteFlag : unsigned int
+	{
+		// SET:     Allow overwritting the file if it exists (mutually exclusive from Append).
+		// NOT SET: Do not allow overwritting an exisitng file.
+		Overwrite = 0b0000'0001,
```

Comment:
Is this maybe just asking for Write permission? I don't believe this is a simple portable way to distinguish between write access for new data (being appended) versus overwriting of existing data.